### PR TITLE
Update MySQL JDBC driver to 8.2.0

### PR DESCRIPTION
**Pull request by Akanksha-kedia**
## Description
This update involves upgrading the MySQL JDBC driver to version 8.2.0.

## Motivation and Context

Note: This artifact was moved to:

[com.mysql](https://mvnrepository.com/artifact/com.mysql) » [mysql-connector-j](https://mvnrepository.com/artifact/com.mysql/mysql-connector-j)

MySQL Connector/J artifacts moved to reverse-DNS compliant Maven 2+ coordinates.

image

The upgrade is necessary due to the discovery of direct vulnerabilities in the earlier version of the driver, as listed below:
Direct vulnerabilities:
[CVE-2022-21363](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-21363)
[CVE-2019-2692](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-2692)
[CVE-2018-3258](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-3258)

## Impact
The component impacted by this update is `presto-mysql`. However, the impact is not significant and should not affect the overall functionality or performance.

## Test Plan
Ran UT CASES 
image
Screenshot 2023-12-14 at 4 12 07 PM
Screenshot 2023-12-14 at 4 12 27 PM

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

== NO RELEASE NOTE ==

### CODE REVIEW

**Comment by Akanksha-kedia** [comment]
@jaystarshot @agrawalreetika please review.

**Comment by steveburnett** [comment]
Please fix the formatting of the release note entry, following the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines)

**Comment by tdcmeehan** [comment]
Please address the build and test failures

**Comment by Akanksha-kedia** [comment]
@tdcmeehan please review.

File: presto-benchmark-runner/pom.xml [hunk 1]
```diff
@@ -177,6 +183,12 @@
             <groupId>com.facebook.presto</groupId>
             <artifactId>testing-mysql-server-5</artifactId>
             <scope>test</scope>
+            <exclusions>
```
Comment (tdcmeehan) [inline presto-benchmark-runner/pom.xml:186]: Does this mean the tests are still running with MySQL server version 5?

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -1178,9 +1178,9 @@
             </dependency>
 
             <dependency>
-                <groupId>mysql</groupId>
-                <artifactId>mysql-connector-java</artifactId>
-                <version>5.1.48</version>
+                <groupId>com.mysql</groupId>
+                <artifactId>mysql-connector-j</artifactId>
+                <version>8.1.0</version>
```
Comment (agrawalreetika) [inline pom.xml:1183]: I see latest version in `8.2.0` [here](https://mvnrepository.com/artifact/com.mysql/mysql-connector-j) 
Any reason, we are not moving to that?

Comment (Akanksha-kedia) [reply pom.xml:1183]: No specific reason i can try with 8.2.0 as well.

File: presto-benchmark-runner/pom.xml (hunk 1 above)
Comment (Akanksha-kedia) [reply presto-benchmark-runner/pom.xml:186]: The scope of this dependency is set to test, indicating that it's only for the test environment. answering to ur question: yes

Comment (tdcmeehan) [reply presto-benchmark-runner/pom.xml:186]: I think we need a new version of this dependency for version 8 so our tests and prod are consistent.

Comment (Akanksha-kedia) [reply presto-benchmark-runner/pom.xml:186]: I was looking for the same but could not find other similar dependency.

Comment (tdcmeehan) [reply presto-benchmark-runner/pom.xml:186]: We'll need to create it, the source code is here: https://github.com/prestodb/testing-mysql-server

**Comment by Akanksha-kedia** [comment]
Yes will check on this further.

**Comment by tdcmeehan** [comment]
Please reopen when the dependency problem is addressed, thanks!

File: presto-benchmark-runner/pom.xml (hunk 1 above)
Comment (Akanksha-kedia) [reply presto-benchmark-runner/pom.xml:186]: cloned the source repo and was trying to build 

biut facing this even after installing the required libraries, Can someone help with this?

**Pull request closed without merging**